staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter_test.go

```diff
@@ -179,3 +202,71 @@ func TestRoundTrip(t *testing.T) {
 		}
 	}
 }
+
+func doUnrecognized(t *testing.T, jsonData string, item runtime.Object) {
+	unstr := make(map[string]interface{})
+	err := json.Unmarshal([]byte(jsonData), &unstr)
+	if err != nil {
+		t.Errorf("Error when unmarshaling to unstructured: %v", err)
+		return
+	}
+	// Now we have unstructured representation of the data.
+
+	data, err := json.Marshal(unstr)
+	if err != nil {
+		t.Errorf("Error when marshaling unstructured: %v", err)
+		return
+	}
+	unmarshalledObj := reflect.New(reflect.TypeOf(item).Elem()).Interface()
+	err = json.Unmarshal(data, &unmarshalledObj)
+	if err != nil {
+		t.Errorf("Error when unmarshaling to object: %v", err)
+		return
+	}
+
+	newObj := reflect.New(reflect.TypeOf(item).Elem()).Interface().(runtime.Object)
+	err = NewConverter().FromUnstructured(unstr, newObj)
+	if err != nil {
+		t.Errorf("FromUnstructured failed: %v", err)
+		return
+	}
+
+	if !reflect.DeepEqual(unmarshalledObj, newObj) {
+		t.Errorf("Object changed, diff: %v", diff.ObjectReflectDiff(unmarshalledObj, newObj))
+	}
+}
+
+func TestUnrecognized(t *testing.T) {
+	testCases := []struct{
+		data string
+		obj  runtime.Object
+	}{
+		{
+			data: "{\"da\":[3.0,\"3.0\",null]}",
```

Comment:
OK - I hope that this time I understood correctly.
I changed the test to check if:
1) serialized json -> object
2) serialized json -> map[string]interface{} -> object
produce the same output

I also added a bunch of other test cases.

I hope that this time is what you wanted.